adcm_client/objects.py

```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
 
     @allure_step("Save config")
     def config_set(self, data):
+        config = self.config(full=True)
         if "config" in data and "attr" in data:
             # We are in a new mode with full_info == True
             if data["attr"] is None:
-                data["attr"] = []
-            history_entry = self._subcall(
-                'config', 'history', 'create',
-                config=data["config"],
-                attr=data["attr"],
-            )
+                data["attr"] = {}
+            history_entry = self._subcall('config', 'history', 'create', **data)
             return history_entry
-        history_entry = self._subcall('config', 'history', 'create', config=data)
+        history_entry = self._subcall(
+            'config', 'history', 'create', config=data, attr=config['attr'])
         return history_entry['config']
 
     @allure_step("Save config")
-    def config_set_diff(self, data):
-        config = self.config()
-        for gk, gv in data.items():
-            if isinstance(gv, dict):
-                for k, v in gv.items():
-                    config[gk][k] = v
-            else:
-                config[gk] = gv
-        self.config_set(config)
+    def config_set_diff(self, data, full=False):
+
+        def update(d, u):
+            for key, value in u.items():
+                if isinstance(value, abc.Mapping):
+                    d[key] = update(d.get(key, {}), value)
```

Comment:
I believe that you could use dict.update(another_dict) function as part of this merge procedure. See https://docs.python.org/3/library/stdtypes.html#dict.update

Reply:
I think that we need a deep update, and the built-in method cannot do this.